pkg/internal/itest/http_fetch_test.go

```diff
@@ -0,0 +1,574 @@
+package itest
+
+import (
```

Comment:
not a now problem, but for the future, if we seperate the HTTP daemon, we'll want to make sure not to lose all these tests. I wonder given plans if we should consider at some point converting this to direct fetch. At the same time, HTTP API is the thing we're delivering for now, and I'd rather we focus our testing on that.